### PR TITLE
Enable install of multiple specs from yaml files

### PR DESCRIPTION
**Pull request by nazavode**
This enables multiple `--file` options for `install`:

```
$ spack install -f spec1.yaml  -f spec2.yaml -f spec3.yaml
```

Allows to mix specs from cli and `yaml` files at the same time:

```
$ spack install -f spec1.yaml  -f spec2.yaml qt mpi graphviz
```

During the streamlining of arguments handling logic in `install`, #6983 has been fixed. Added regressions for it, and tests for the new feature.

### CODE REVIEW

File: lib/spack/spack/test/cmd/install.py [hunk 1]
```diff
@@ -259,10 +259,41 @@ def test_install_from_file(spec, concretize, error_code, tmpdir):
     if concretize:
         spec.concretize()
 
-    with fs.working_dir(str(tmpdir)):
+    specfile = tmpdir.join('spec.yaml')
+
+    with specfile.open('w') as f:
+        spec.to_yaml(f)
+
+    # Relative path to specfile (regression for #6906)
```
Comment (alalazo) [inline lib/spack/spack/test/cmd/install.py]: Quite nice idea. We should probably take the habit of noting down the issue if we implement a test for a regression.

Comment (nazavode) [reply lib/spack/spack/test/cmd/install.py]: Began doing this after a couple of nightmare debugging nights both ended up in finding out that I was just introducing regressions. Low cost, high return!

Comment (alalazo) [reply lib/spack/spack/test/cmd/install.py]: Cool! I was already thinking of marking slow tests to have something like:
```console
$ spack test -m "not slowtest"
```
running automatically in the background when I am developing. Marking also regressions makes a lot of sense imo.

**Comment by alalazo** [comment]
@nazavode I stopped this CI build. Rebasing on `develop` and pushing should restart a Travis build without the dreaded MacOS tests.

**Comment by nazavode** [comment]
@alalazo thanks, the previous CI build was having a hard time limbing the `macOS` backlog.

**Pull request merged**